scripts/vagrant/optional.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+set -e
+set -x
+
+DB_NAME=${DB_NAME:-"catmaid"}
+DB_USER=${DB_USER:-"catmaid_user"}
+DB_PASSWORD=${DB_PASSWORD:-"p4ssw0rd"}
+if [ -f "$HOME/timezone" ]; then
+    DEFAULT_TZ=$(cat $HOME/timezone)
+else
+    DEFAULT_TZ="America/New_York"
+fi
+TIMEZONE=${TIMEZONE:-$DEFAULT_TZ}
+
+mkdir -p ~/data
+
+echo "Making postgres accessible from host"
+# TODO: make idempotent
```

Comment:
Is idempotence important here? It only protects against someone trying to set up their database repeatedly, in which case they probably have other issues.

Reply:
That's true, but at the same time I don't think it's hard to have edempotence here. To me it looks like only the `postgresql.conf` update might problematic with multiple runs (which are indeed unlikely to happen). To fix this we could just remove all `listening_addresses = ` lines from `postgresql.conf` before we append our own:
```
sudo sed -i "/^\(listening_addresses =\).*/d" $PG_CONF_DIR/postgresql.conf
sudo -c "echo 'listen_addresses = '*''  >> $PG_CONF_DIR/postgresql.conf"
```
Or do i miss something here?

Reply:
No that sounds good! Same with the hba line, a previous iteration grep'd for the line and only added it if it wasn't already there. As `create_superuser` takes user input we can trust them not to try to add the same user multiple times, and I'm wondering whether we should just drop the example projects as they don't resolve to anything - more likely to be a confusion than anything else.

Reply:
I agree, the `root.sh` file looks like the right place for `listening_addresses`. I don't think `pg_hba.conf` is a problem, because the existing file seems to be replaced altogether, so in that sense it is idempotent.

And you are right about there being a danger of having example projects with unreachable data can lead to confusion about whether a setup is working or not. Like catmaid/catmaid#1232 states, it would be nice to have a better example project. We really should use some published neurons with publicly hosted images, but let's continue that discussion in the previously mentioned issue. For the dev setup I think it is still useful to have at least the basic skeleton of a project/stack/project_stack/mirror/etc. setup.